core/src/main/java/hudson/util/Iterators.java

```diff
@@ -305,7 +305,8 @@ public void remove() {
      * Wraps another iterator and throws away nulls.
      */
     public static <T> Iterator<T> removeNull(final Iterator<T> itr) {
-        return com.google.common.collect.Iterators.filter(itr, Predicates.notNull());
+        Iterable<T> iterable = () -> itr;
+        return StreamSupport.stream(iterable.spliterator(), false).filter(t -> t != null).iterator();
```

Comment:
Could use a method reference to `Objects::nonNull` rather than a custom lambda here:

```suggestion
        return StreamSupport.stream(iterable.spliterator(), false).filter(Objects::nonNull).iterator();
```